prime-router/src/main/kotlin/tokens/SenderUtils.kt

```diff
@@ -64,15 +64,29 @@ class SenderUtils {
             val nimbusdsJwk = JWK.parseFromPEMEncodedObjects(pem)
             val jwk = jacksonObjectMapper().readValue(nimbusdsJwk.toJSONString(), Jwk::class.java)
             // All the rest of this is sanity checks
-            val prefix = "Cannot convert pemFile to EC Key. "
-            if (jwk.kty.isNullOrEmpty()) error("$prefix.  Key must have a kty keytype")
-            if (jwk.d != null) error("$prefix This looks like a private key.  Key must be a public key.")
-            if (jwk.x.isNullOrEmpty() || jwk.y.isNullOrEmpty()) error("$prefix. Key missing elliptic point (x,y) value")
-            if (nimbusdsJwk.keyType != KeyType.EC) error("$prefix keyType is  ${nimbusdsJwk.keyType}.  Expecting 'EC'.")
-            // actually generate an ECPublicKey obj, just to confirm it can be done.
-            val ecPublicKey = jwk.toECPublicKey()
-            if (ecPublicKey.w == null) error("$prefix.  'w' Point obj not created")
-            if (ecPublicKey.algorithm != "EC") error("$prefix.  Alg is ${ecPublicKey.algorithm}.  Expecting 'EC'.")
+            if (jwk.kty.isNullOrEmpty()) error("Key must have a kty keytype")
+            if (nimbusdsJwk.keyType == KeyType.EC) {
```

Comment:
This code is pretty redundant with the readPrivateKeyPemFile below.  We should consolidate.   Also, the code below uses "when" which is probably slightly better style than the "if" here.